ts/packages/anchor/src/program/namespace/methods.ts

```diff
@@ -119,15 +119,33 @@ export class MethodsBuilder<IDL extends Idl, I extends AllInstructions<IDL>> {
     accounts: Partial<Accounts<I["accounts"][number]>>
   ): MethodsBuilder<IDL, I> {
     this._autoResolveAccounts = true;
-    Object.assign(this._accounts, accounts);
+    for (const accountName in accounts) {
+      if (accounts[accountName]) {
+        this._accounts[accountName] = accounts[accountName];
+      } else if (
+        accounts[accountName] === null &&
+        this._accountsResolver.isOptional(accountName)
+      ) {
+        this._accounts[accountName] = null;
+      }
+    }
     return this;
   }
 
   public accountsStrict(
     accounts: Accounts<I["accounts"][number]>
   ): MethodsBuilder<IDL, I> {
     this._autoResolveAccounts = false;
-    Object.assign(this._accounts, accounts);
+    for (const accountName in accounts) {
+      if (accounts[accountName]) {
+        this._accounts[accountName] = accounts[accountName];
+      } else if (
```

Comment:
Will this error correctly if `accounts[accountName] === null ** isOptional == false`?

Reply:
It wont error out in runtime (Nor should it IMO), but based on the `Accounts` type change you should fail type checking errors if you try to pass in null into a non optional account (assuming strict mode is on).